src/core/buffer.c

```diff
@@ -114,15 +113,15 @@ void janet_buffer_setcount(JanetBuffer *buffer, int32_t count) {
 
 /* Adds capacity for enough extra bytes to the buffer. Ensures that the
  * next n bytes pushed to the buffer will not cause a reallocation */
-void janet_buffer_extra(JanetBuffer *buffer, int32_t n) {
+void janet_buffer_extra(JanetBuffer *buffer, size_t n) {
     /* Check for buffer overflow */
-    if ((int64_t)n + buffer->count > INT32_MAX) {
+    if (n + buffer->count > JANET_INTMAX_SIZE) {
```

Comment:
We should not remove cast - also needs to be unconditionally 64 bits to avoid overflow issue.